src/System.Reflection.Metadata/src/System/Reflection/Metadata/Ecma335/MetadataSerializer.cs

```diff
@@ -79,7 +79,15 @@ public void SerializeMetadata(BlobBuilder builder, Func<BlobBuilder, ContentId>
 #endif
     sealed class TypeSystemMetadataSerializer : MetadataSerializer
     {
-        private static readonly ImmutableArray<int> EmptyRowCounts = ImmutableArray.CreateRange(Enumerable.Repeat(0, MetadataTokens.TableCount));
+        private static readonly ImmutableArray<int> EmptyRowCounts = CreateEmptyRowCounts();
+
+        private static ImmutableArray<int> CreateEmptyRowCounts()
+        {
+            var builder = ImmutableArray.CreateBuilder<int>(MetadataTokens.TableCount);
+            builder.Count = builder.Capacity;
+            // Memory is already zeroed-out, no need to fill it here
+            return builder.MoveToImmutable();
+        }
```

Comment:
I'm not convinced it's worth spending LOC on this once per app-domain initialization.